compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java

```diff
@@ -3302,11 +3312,29 @@ private boolean isInIsolatedObjectMethod(SymbolEnv env, boolean ignoreInit) {
             return false;
         }
 
-        BType ownerType = Types.getImpliedType(enclInvokable.symbol.owner.type);
+        BType ownerType = Types.getImpliedType(enclFunction.symbol.owner.type);
 
         return ownerType.tag == TypeTags.OBJECT && isIsolated(ownerType.flags);
     }
 
+    private BLangFunction getEnclNonAnonymousFunction(BLangFunction enclFunction) {
+        if (enclFunction == null || enclFunction.symbol.owner.kind == SymbolKind.PACKAGE) {
+            return null;
+        }
+
+        if (!Symbols.isFlagOn(enclFunction.symbol.flags, Flags.ANONYMOUS)) {
+            return enclFunction;
+        }
+
+        BLangNode enclNode = enclFunction.parent;
+
+        while (enclNode != null && enclNode.getKind() != NodeKind.FUNCTION) {
```

Comment:
When looking at the following example, it compiles without any errors. Here, we need to check the node kind to determine whether it is a resource function.
```ballerina
client isolated  class TestEP {
    private int[][] arr = [];           
    resource function get hello(string name) {
        function v = function (int[] node) {
            self.arr.push(node);
        };
    }
}
```

Reply:
Fixed in 4348c4b, thanks!